modules/st2-api/api.js

```diff
@@ -255,7 +256,8 @@ export class API {
     return new Promise((resolve, reject) => {
       try {
         const source = new EventSource(streamUrl, {
-          rejectUnauthorized: this.rejectUnauthorized,
+          https: {rejectUnauthorized: this.rejectUnauthorized},
+          headers: {'X-Auth-Token': this.token.token},
```

Comment:
:100:

That's a nice assembled PR! :+1: 
I bet it took quite of time to debug and catch it .

---

FYI, some history that could be related or at least good to know: https://github.com/StackStorm/st2/pull/3566

Reply:
Thanks Eugen. Yes it took a while to get a handle on the issue. Now it is taking long time to figure out the CircleCI tests failing